File: Javascript/Solved/167_TwoSumII.js
/**
 * @param {number[]} numbers
 * @param {number} target
 * @return {number[]}
 */


var twoSum = function(numbers, target) {
    
    
    var s = 0, e = numbers.length-1;
    
    while (s < e) {
        var sum = numbers[s] + numbers[e];
        if (sum === target) {
            return [s + 1, e + 1];
        } else if (sum < target) {
            s++;
        } else {
            e--;
        }
    }
    
    return [-1, -1];
};

File: Golang/Solved/167_TwoSumII.go
package main

func twoSum(numbers []int, target int) []int {

	s, e := 0, len(numbers)-1

	for numbers[s]+numbers[e] != target {

		sum := numbers[s] + numbers[e]

		if sum < target {
			s++
		} else if sum > target {
			e--
		} else {
			return []int{s + 1, e + 1}
		}
	}

	return []int{}

}
